connectors/dexory/connector.py

```diff
@@ -0,0 +1,341 @@
+        if page_count % 10 == 0:
+            log.info(f"Processed {page_count} pages")
+
+        # Check for next page
+        links = response_data.get("links", {})
+        next_url = links.get("next")
+
+    return page_count, location_count
+
+
+def update(configuration: dict, state: dict) -> None:
+    """
+     Define the update function, which is a required function, and is called by Fivetran during each sync.
+    See the technical reference documentation for more details on the update function
+    https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+    Args:
+        configuration: A dictionary containing connection details
+        state: A dictionary containing state information from previous runs
+        The state dictionary is empty for the first sync or for any full re-sync
+    """
+    log.warning("Example: Dexory")
+
+    # Validate the configuration to ensure it contains all required values.
+    validate_configuration(configuration=configuration)
+
+    # Get configuration values
+    site_name = configuration.get("site_name")
+    api_key = configuration.get("site_api_key")
+    base_url = configuration.get("base_url", __DEFAULT_BASE_URL)
+
+    log.info(
+        f"Max pages: {__DEFAULT_MAX_PAGES}, Page size: {__DEFAULT_PAGE_SIZE}, Base retry delay: {__BASE_RETRY_DELAY_SECONDS}s"
+    )
+    log.info(f"Syncing data for site: {site_name}")
+
+    try:
+        page_count, location_count = _sync_locations_pages(
+            site_name,
+            api_key,
+            __DEFAULT_PAGE_SIZE,
+            base_url,
+            __DEFAULT_MAX_PAGES,
+            __BASE_RETRY_DELAY_SECONDS,
+        )
+
+        log.info(f"Completed: {page_count} pages, {location_count} locations")
+
+        # Save the progress by checkpointing the state. This is important for ensuring that the sync process can resume
+        # from the correct position in case of next sync or interruptions.
+        # Learn more about how and where to checkpoint by reading our best practices documentation
+        # (https://fivetran.com/docs/connectors/connector-sdk/best-practices#largedatasetrecommendation).
+        new_state = {"last_sync": datetime.now(timezone.utc).isoformat()}
+        op.checkpoint(state=new_state)
+
+    except Exception as e:
+        log.severe(f"Sync failed: {e}")
+        raise
+
+    log.info("Dexory connector sync completed successfully")
+
+
+def schema(configuration: dict) -> List[Dict[str, Any]]:
+    """
+    Define the schema function which lets you configure the schema your connector delivers.
+    See the technical reference documentation for more details on the schema function:
+    https://fivetran.com/docs/connectors/connector-sdk/technical-reference#schema
+    Args:
+        configuration: a dictionary that holds the configuration settings for the connector.
+    """
+    return [
+        {"table": "location", "primary_key": ["name", "location_type", "aisle", "scan_date"]},
```

Comment:
```suggestion
        {
            "table": "location", 
            "primary_key": ["name", "location_type", "aisle", "scan_date"]
        },
```
can we format this in this way to make it more readable